### PR TITLE
feat(cognito): import an existing user pool client

### PR DESCRIPTION
**Pull request by nija-at**
### Commit Message

Include other clean up around awslint exclusions and CloudFormation
resource attributes.

BREAKING CHANGE: `UserPoolClient` construct no longer has the property
`userPoolClientClientSecret`. The functionality to retrieve the client
secret never existed in CloudFormation, so this property was not
working in the first place.
* **cognito:** The `userPoolClientName` property on the `UserPoolClient`
construct will throw an error if client name was not configured on the
`UserPoolClient` during initialization. This property was previously
incorrectly configured and was returning a not-implemented message from
CloudFormation every time.
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cognito/package.json [hunk 1]
```diff
@@ -90,10 +90,8 @@
   },
   "awslint": {
     "exclude": [
-      "docs-public-apis:@aws-cdk/aws-cognito.UserPoolClient.userPoolClientClientSecret",
-      "docs-public-apis:@aws-cdk/aws-cognito.UserPoolClient.userPoolClientId",
-      "docs-public-apis:@aws-cdk/aws-cognito.UserPoolClient.userPoolClientName",
-      "docs-public-apis:@aws-cdk/aws-cognito.UserPoolClientProps"
+      "attribute-tag:@aws-cdk/aws-cognito.UserPoolClient.userPoolClientName",
+      "resource-attribute:@aws-cdk/aws-cognito.UserPoolClient.userPoolClientClientSecret"
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/package.json]: Take a look at the note here - https://github.com/aws/aws-cdk/pull/7091/files#diff-112b6222454a4f23b78d56cd0fa66b95R88-R92 - on why these two are needed.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts [hunk 2]
```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
...
-  public readonly userPoolClientName: string;
+  public static fromUserPoolClientId(scope: Construct, id: string, userPoolClientId: string): IUserPoolClient {
+    class Import extends Resource implements IUserPoolClient {
+      public readonly userPoolClientId = userPoolClientId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly userPoolClientId: string;
```
Comment (eladb) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: doc?

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts [hunk 3]
```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
...
 
   /**
-   * @attribute
+   * The client name that was specified via the `userPoolClientName` property during initialization,
+   * `undefined` otherwise.
+   */
+  public readonly userPoolClientName?: string;
```
Comment (eladb) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: I would expect this to be an attribute and never `undefined`, no?

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts [hunk 4]
```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
...
+   * at this time, please use the CLI or Console to retrieve this value".
+   * Awaiting updates from CloudFormation.
    */
-  public readonly userPoolClientClientSecret: string;
 
   constructor(scope: Construct, id: string, props: UserPoolClientProps) {
-    super(scope, id, {
-      physicalName: props.userPoolClientName,
-    });
+    super(scope, id);
```
Comment (eladb) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: Why are we not using physical name?

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts [hunk 5]
```diff
@@ -42,44 +45,64 @@ export interface UserPoolClientProps {
...
+      clientName: props.userPoolClientName,
       generateSecret: props.generateSecret,
       userPoolId: props.userPool.userPoolId,
       explicitAuthFlows: props.enabledAuthFlows
     });
 
     this.userPoolClientId = resource.ref;
-    this.userPoolClientClientSecret = resource.attrClientSecret;
-    this.userPoolClientName = resource.attrName;
+    this.userPoolClientName = props.userPoolClientName;
```
Comment (eladb) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: I would expect to call super.getResourceName()

**Comment by eladb** [comment]
added "do-not-merge"

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 2 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: Automatically inherited from the interface, I believe. Hence the no complaints from `awslint`.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 3 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: No, does not work. CloudFormation returns the message "Name attribute not supported at this time, please use the CLI or Console to retrieve this value"

See my comment here - https://github.com/aws/aws-cdk/pull/7091/files#diff-112b6222454a4f23b78d56cd0fa66b95R88-R92

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 4 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: User pool clients generate their own primary identifiers (e.g. `6tapehhff4ov922cjnheqpfr23`) that cannot be specified as input. AFAIU, physical names are set for resources whose primary identifiers can be configured by the user. Am I wrong?

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 3 above)
Comment (eladb) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: Then I would implement this as a getter that throws an error if the name is not provided but at least we get a clean api that we can evolve later to return a token.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 4 above)
Comment (eladb) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: Okay makes sense. Maybe add a comment.

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 3 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts]: Good point

**Pull request merged**